engine/src/main/java/org/terasology/rendering/dag/stateChanges/EnableMaterial.java

```diff
@@ -25,7 +25,7 @@
 /**
  * TODO: Add javadocs
  */
-public class EnableMaterial implements StateChange {
+public final class EnableMaterial implements StateChange {
```

Comment:
But what was the reasoning behind the first final classes then? I didn't notice them before and I wonder why we should close the door to further subclassing. I can hardly imagine why anybody _would_ subclass these as they are pretty specialized already. But I guess one never knows?

I've done a quick search on the subject and some people advocate that a class should either be designed to be extensible OR declared final, mostly as a form of defensive programming. I don't think we use this rule in Terasology's codebase though. So I'm not sure this is necessary but it's ok to leave it like this for the time being. I'm just curious if there was a good reason for this.